sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightVersionsPlugin.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.sqldelight.gradle
+
+import app.cash.sqldelight.VERSION
+import org.gradle.api.Plugin
+import org.gradle.api.initialization.Settings
+
+abstract class SqlDelightVersionsPlugin : Plugin<Settings> {
+  override fun apply(settings: Settings) {
+    settings.dependencyResolutionManagement.versionCatalogs.register("sqldelight") {
+      val version = it.version("sqldelight", VERSION)
+      it.plugin("sqldelight", "app.cash.sqldelight").versionRef(version)
+
+      fun include(name: String, artifact: String = name) {
+        it.library(name, "app.cash.sqldelight", artifact).versionRef(version)
+      }
+
+      include("adapters-primitive", "primitive-adapters")
```

Comment:
Here's my spicy idea: we could generate an Enum at build time (similar to the `VERSION` constant) that lists every published gradle module in this project. Then we could use an exhaustive `when` here to map those Enum constants to the version catalog string. That way we won't ever accidentally forget to include a new module.

Reply:
Yeah, but you need to iterate over all projects and parse each properties file too. And some projects are internal, like test-utils or intellij plugin. Ideally, there is some built-in function in Gradle...

Reply:
I personally think it'd still be worthwhile, but either way it's best left for a separate PR!